### DISCUSSION TITLE
Integration with other editors

### DISCUSSION THREAD

**Discussion by elken** [community/community#16835] [Codespaces] [answered]
Hi all,

Codespaces looks great! Could be a winner for sure, it comes off as quite similar to [Gitpod](https://www.gitpod.io/) which also has the advantage of integrating with other editors (currently just WIP emacs, which is my editor of choice, and future plans of jetbrains).

Would this be something on the roadmap for codespaces too, or is this going to forever be VSCode only?

**Comment by dmilburn** [comment, marked as answer]
Hi from the Codespaces team! Glad you're liking Codespaces so far 😄  an existing alternative we have to VSCode is you're able to SSH into a codespace using the gh cli: https://cli.github.com/manual/gh_codespace_ssh. I don't have enough familiarity with Emacs to say for sure whether it would work without digging in, but I know lots of folks on our team are able to use Vim to edit their codespace this way.

Being able to use other editors is definitely something on our roadmap. Stay tuned for more updates!

**Reply by elken** [reply]
Hi Debbie, thanks for a response!

I wasn't aware of this, that should help!

> Being able to use other editors is definitely something on our roadmap

That's encouraging! I appreciate that VSCode is the environment it's designed around but glad to see there are plans for more :)

**Reply by Anselmoo** [reply]
@dmilburn thx for your detailed answer, I am wondering if `Codespaces` also support ftp-clients like [ForkLift](https://binarynights.com) via ssh?
In the case of up and downloading benchmark sets, this might be helpful to exclude unnecessary commits.

thx
